src/bin/ambit/cmd.rs

```diff
@@ -0,0 +1,191 @@
+        return Err(AmbitError::Other(
+            "Dotfile repository does not exist. Run `init` or `clone` before syncing.".to_owned(),
+        ));
+    }
+    let mut successful_symlinks = 0; // Number of symlinks that actually occurred
+    let mut total_symlinks = 0;
+    // link is a closure as it must have access to outside variables
+    let mut link = |repo_filename: &str, host_filename: &str| -> AmbitResult<()> {
+        let host_file = AmbitPath::new(
+            AMBIT_PATHS.home.path.join(host_filename),
+            AmbitPathKind::File,
+        );
+        let host_file_str = host_file.to_str()?;
+        let repo_file = AmbitPath::new(
+            AMBIT_PATHS.repo.path.join(repo_filename),
+            AmbitPathKind::File,
+        );
+        let repo_file_str = repo_file.to_str()?;
+
+        // already_symlinked holds whether host_file already links to repo_file
+        let already_symlinked = if let Ok(link_path) = fs::read_link(&host_file.path) {
+            link_path == repo_file.path
+        } else {
+            false
+        };
+
+        if host_file.exists() && !already_symlinked {
+            // Host file already exists but is not symlinked correctly
+            return Err(AmbitError::Symlink {
+                host_file: host_file_str.to_owned(),
+                repo_file: repo_file_str.to_owned(),
+                error: Box::new(AmbitError::Other(
+                    "Host file already exists and is not correctly symlinked".to_owned(),
+                )),
+            });
+        } else if !repo_file.exists() {
+            return Err(AmbitError::Symlink {
+                host_file: host_file_str.to_owned(),
+                repo_file: repo_file_str.to_owned(),
+                error: Box::new(AmbitError::Other(
+                    "Repository file does not exist".to_owned(),
+                )),
+            });
+        } else if !already_symlinked {
+            if !dry_run {
+                // Attempt to perform symlink
+                if let Err(e) = symlink(&repo_file.path, &host_file.path) {
+                    // Symlink went wrong
+                    return Err(AmbitError::Symlink {
+                        host_file: host_file_str.to_owned(),
+                        repo_file: repo_file_str.to_owned(),
+                        error: Box::new(AmbitError::Io(e)),
+                    });
+                }
+                successful_symlinks += 1;
+            }
+            if !quiet {
+                println!("{} -> {}", host_file_str, repo_file.to_str()?);
+            }
+        }
+        total_symlinks += 1;
+        Ok(())
+    };
+    let entries = get_config_entries()?;
+    for entry in entries {
+        let left_specs = entry.left.into_iter();
+        match entry.right {
+            Some(right_specs) => {
+                let right_specs = right_specs.into_iter();
+                for path_pair in left_specs.zip(right_specs) {
```

Comment:
You can skip the `path_pair` variable and simply do `for (repo_file, host_file) in`.